src/main/java/org/springframework/samples/petclinic/owner/OwnerController.java

```diff
@@ -158,6 +178,57 @@ public String processFindForm(Owner owner, BindingResult result, Map<String, Obj
             return "owners/ownersList";
         }
     }
+    
+    @GetMapping("/owners_first_name")
+    public String findOwnersByFirstName(Owner owner, BindingResult result, Map<String, Object> model, HttpServletRequest request) {
+
+        // allow parameterless GET request for /owners to return all records
+        if (owner.getFirstName() == null || owner.getFirstName().isEmpty()) {
+            owner.setFirstName(""); // empty string signifies broadest possible search
+            
+            // logging use of listOfOwners page when accessed old way
+            if (PetClinicToggles.toggleListOfOwners.isOn()) {
+                listOfOwnerCsvLogger.info(request.getRemoteAddr() + ",1");
+            } else {
+                listOfOwnerCsvLogger.info(request.getRemoteAddr() + ",0");
+            }
+            
+            
+            if (PetClinicToggles.toggleFindOwnerByFirstName.isOn()) {
```

Comment:
Don't log your stuff inside if, it will log twice. You only care that the search bar was used